app/controllers/report_controller/reports.rb

```diff
@@ -89,11 +89,6 @@ def miq_report_delete
       end
     else
       begin
-        if rpt.rpt_type == "Default"
-          add_flash(_("Default %{model} \"%{name}\" cannot be deleted") % {:model=>ui_lookup(:model=>"MiqReport"), :name=>rpt.name}, :error)
-          redirect_to :action=>"editreport", :id=>rpt.id, :flash_msg=>flash, :flash_error=>true
-          return
-        end
         rpt_name = rpt.name
         audit = {:event=>"report_record_delete", :message=>"[#{rpt_name}] Record deleted", :target_id=>rpt.id, :target_class=>"MiqReport", :userid => session[:userid]}
         rpt.destroy
```

Comment:
My concern here: are not we just assuming that the check here is done on the client side? I mean, what if I hack the request in browser to delete a "Default" report?

I *think* that we should here either ignore the request to delete such report or raise an exception (that has to be properly handled so that we do not dump stack trace to the user ofc)